### PR TITLE
ref(server): Add a dedicated metric for transaction usage

### PR DESCRIPTION
**Pull request by jan-auer**
Adds a dedicated and cheap counter metric for usage tracking. It contains the `has_profile` tag, which will later be removed from the durations metric to lower its cardinality.

This metric should improve the performance of all usage queries. In the long term, it allows to decouple product metrics from bookkeeping.

### CODE REVIEW

**Comment by jan-auer** [comment]
This is ready for review while I fix integration tests.

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 1]
```diff
@@ -322,13 +319,24 @@ impl TransactionExtractor<'_> {
...
         // Duration
+        let duration = relay_common::time::chrono_to_positive_millis(end - start);
         metrics.project_metrics.push(
             TransactionMetric::Duration {
                 unit: DurationUnit::MilliSecond,
-                value: relay_common::time::chrono_to_positive_millis(end - start),
+                value: duration,
                 tags: TransactionDurationTags {
-                    has_profile: self.has_profile,
+                    has_profile: self.has_profile, // TODO: Stop producing this in v3
```
Comment (iker-barriocanal) [inline relay-server/src/metrics_extraction/transactions/mod.rs:339]: Are we still producing this tag until we update the consumers?

File: relay-server/src/utils/metrics_rate_limits.rs [hunk 2]
```diff
@@ -38,7 +38,12 @@ impl<Q: AsRef<Vec<Quota>>> MetricsLimiter<Q> {
     /// Create a new limiter instance.
     ///
     /// Returns Ok if `metrics` contain transaction metrics, `metrics` otherwise.
-    pub fn create(buckets: Vec<Bucket>, quotas: Q, scoping: Scoping) -> Result<Self, Vec<Bucket>> {
+    pub fn create(
+        buckets: Vec<Bucket>,
+        quotas: Q,
+        scoping: Scoping,
+        usage: bool,
```
Comment (iker-barriocanal) [inline relay-server/src/utils/metrics_rate_limits.rs:45]: `usage` may be hard to understand outside this scope. What do you think about making this an enum of two variants (usage and transaction duration)?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-server/src/metrics_extraction/transactions/mod.rs (hunk 1 above)
Comment (jjbayer) [reply relay-server/src/metrics_extraction/transactions/mod.rs:339]: We have to.

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/metrics_extraction/transactions/mod.rs (hunk 1 above)
Comment (jan-auer) [reply relay-server/src/metrics_extraction/transactions/mod.rs:339]: There's no functional need to remove the tag, everything works correctly with and without the tag, but as @jjbayer pointed out we need to keep it until the consumers are fully updated.

File: relay-server/src/utils/metrics_rate_limits.rs (hunk 2 above)
Comment (jan-auer) [reply relay-server/src/utils/metrics_rate_limits.rs:45]: Agreed. The metrics rate limiter overall has a convoluted signature, so I chose the simplest change to follow up with a proper refactoring. I'm not yet sure about whether this should be an enum, the config, or something else, so I'd prefer to stick with the simplest choice. Would that work?

**Pull request merged**